cudaextender/samples/sample_cudaextender.cpp

```diff
@@ -0,0 +1,256 @@
+/*
+* Copyright 2020 NVIDIA CORPORATION.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+#include <file_location.hpp>
+#include <claraparabricks/genomeworks/cudaextender/extender.hpp>
+#include <claraparabricks/genomeworks/io/fasta_parser.hpp>
+#include <claraparabricks/genomeworks/utils/cudautils.hpp>
+#include <claraparabricks/genomeworks/utils/device_buffer.hpp>
+#include <claraparabricks/genomeworks/utils/pinned_host_vector.hpp>
+
+#include <iostream>
+#include <string>
+#include <vector>
+#include <fstream>
+#include <cuda.h>
+
+using namespace claraparabricks::genomeworks;
+using namespace cudautils;
+using namespace cudaextender;
+
+constexpr char A_NT = 0;
+constexpr char C_NT = 1;
+constexpr char G_NT = 2;
+constexpr char T_NT = 3;
+constexpr char L_NT = 4;
+constexpr char N_NT = 5;
+constexpr char X_NT = 6;
+constexpr char E_NT = 7;
+constexpr char NUC  = 8;
+constexpr char NUC2 = NUC * NUC;
+
+// Really simple parser with no error checks
+void parse_seed_pairs(const std::string& filepath, std::vector<SeedPair>& seed_pairs)
+{
+    std::ifstream seed_pair_file(filepath);
+    if (!seed_pair_file.is_open())
+        throw std::runtime_error("Cannot open file");
+    if (seed_pair_file.good())
+    {
+        std::string line;
+        while (std::getline(seed_pair_file, line, ','))
+        {
+            SeedPair seed_pair;
+            seed_pair.target_position_in_read = std::atoi(line.c_str());
+            std::getline(seed_pair_file, line); // Get the next value
+            seed_pair.query_position_in_read = std::atoi(line.c_str());
+            seed_pairs.push_back(seed_pair);
+        }
+    }
+}
+
+// convert input sequence from alphabet to integers
+void encode_string(char* dst_seq, const char* src_seq, int32_t len)
+{
+    for (int32_t i = 0; i < len; i++)
+    {
+        char ch = src_seq[i];
+        char dst;
```

Comment:
this variable is unused and `ch` can be `const`.

Reply:
Done.